cilium-cli/connectivity/tests/encryption.go

```diff
@@ -197,6 +197,26 @@ func isWgEncap(t *check.Test) bool {
 	return true
 }
 
+// checkIPSecPodToPod checks whether in case of IPSec being enabled and used
+// during the podToPodEncryption test it should be executed or skipped due to:
+//
+// 1. missing backporting of the commit in previous versions;
+// 2. usage of IPv6, for which the test is flaky (see https://github.com/cilium/cilium/issues/35485).
+//
+// Once the above reasons are fixed, this function can be removed.
+func checkIPSecPodToPod(t *check.Test, ipFam features.IPFamily) error {
+	if e, ok := t.Context().Feature(features.EncryptionPod); !(ok && e.Enabled && e.Mode == "ipsec") {
+		return nil
+	}
+	if !versioncheck.MustCompile(">=1.17.0")(t.Context().CiliumVersion) {
+		return fmt.Errorf("enabling test for IPSec requires backporting")
+	}
```

Comment:
feature and version checks can be put in cilium-cli/connectivity/builder/pod_to_pod_encryption.go to skip the whole test at all with clear skip reason. Like what you did at https://github.com/cilium/cilium/pull/35486/files#diff-7a17abdbd1cb8e727bf901ec2001452944a48e4d03d83a46ee7e8e760fe9e6b3R30-R39

Reply:
Ok, but then I don't think the following is enough [here](https://github.com/cilium/cilium/blob/main/cilium-cli/connectivity/tests/encryption.go#L347), as it would prevent also WireGuard+IPv6:

```go
if ipFam == features.IPFamilyV6 {
    // comment, #issue_no
    return
}
```

I would need to add:

```go
if f, ok := t.Context().Feature(features.EncryptionPod); ok && f.Enabled && f.Mode == "ipsec" && ipFam == features.IPFamilyV6 {
    // comment, #issue_no
    return
}
		
```

and that is why I decided to create a unique function where to check both the requirements (cilium version and ipFamily with IPSec).

Reply:
You are correct. Resolve the conversation.

Reply:
Let me know what is the preferred way, I'm ok with both. If we want to split the checks between the two files (version in `pod_to_pod_encryption.go` and IPSec+IPv6 in `encryption.go`) I'll correct the last commit fix :+1: